### PR TITLE
Kotlin migration DeckSelectionDialog

### PR DESCRIPTION
**Pull request by Arnold2381**
Kotlin migration DeckSelectionDialog

### CODE REVIEW

**Review by mikehardy — APPROVED** [review]
All seems fine - thanks

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt [hunk 1]
```diff
@@ -13,511 +13,395 @@
...
+            Timber.w(ex)
         }
     }
 
-    private void showDeckDialog() {
-        CreateDeckDialog createDeckDialog =  new CreateDeckDialog(requireActivity(), R.string.new_deck, CreateDeckDialog.DeckDialogType.DECK, null);
-        createDeckDialog.setOnNewDeckCreated((id) -> {
+    private fun showDeckDialog() {
+        val createDeckDialog = CreateDeckDialog(requireActivity(), R.string.new_deck, CreateDeckDialog.DeckDialogType.DECK, null)
+        createDeckDialog.setOnNewDeckCreated { id: Long? ->
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt:151]: @Arnold2381 Could you flag this method for a follow-up to make the parameter `@NonNull` and check that it currently doesn't return null

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt [hunk 2]
```diff
@@ -13,511 +13,395 @@
...
 
+        init {
+            mAllDecksList.addAll(deckNames)
+            mCurrentlyDisplayedDecks.addAll(deckNames)
+            mCurrentlyDisplayedDecks.sort()
+        }
+    }
 
-    public static class SelectableDeck implements Comparable<SelectableDeck>, Parcelable {
+    open class SelectableDeck : Comparable<SelectableDeck>, Parcelable {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt:308]: Add a `@KotlinCleanup` to auto-parcel this

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Minor changes

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt (hunk 1 above)
Comment (Arnold2381) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt:154]: `setOnNewDeckCreated ` doesnt return null.

**Review by david-allison-1 — APPROVED** [review]
Gogo

**Pull request merged**